st2-upstart/bin/setup_container.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+ST2USER=$1
+ST2PASSWORD=$2
+
+htpasswd -bs /etc/st2/htpasswd ${ST2USER} ${ST2PASSWORD}
+st2 login -p ${ST2PASSWORD} ${ST2USER}
+
+echo "export ST2USER='${ST2USER}'" > ~root/st2.vars
+echo "export ST2PASSWORD='${ST2PASSWORD}'" >> ~root/st2.vars
+echo "export ST2_AUTH_TOKEN='`st2 auth -t -p ${ST2PASSWORD} ${ST2USER}`'" >> ~root/st2.vars
```

Comment:
The reason I did it this way is that `st2-self-check` doesn't respect the `~/.st2/config` and `~/.st2/token-username` files. It requires the `$ST2_AUTH_TOKEN` environment variable to be specified. See https://github.com/StackStorm/st2/issues/3295 for more info. If we don't need to use `st2-self-check`, then I suspect this may be a non-issue, at least as it relates to `st2-docker`.